### PR TITLE
Use options.remotingContext for conditional settings

### PR DESCRIPTION
**Pull request by raymondfeng**
### Description

This PR uses `options` from remote requests to decide default values for certain security-related settings

#### Related issues

See https://github.com/strongloop/loopback-datasource-juggler/pull/1638#issuecomment-437407868
https://github.com/strongloop/loopback-datasource-juggler/issues/1663

- connect to <link_to_referenced_issue>

### Checklist

- [x] New tests added or existing tests modified to cover all changes
- [x] Code conforms with the [style
  guide](http://loopback.io/doc/en/contrib/style-guide.html)

### CODE REVIEW

File: lib/connectors/memory.js [hunk 1]
```diff
@@ -345,7 +345,7 @@ Memory.prototype.findOrCreate = function(model, filter, data, callback) {
     });
   }
 
-  self._models[model].model.include(nodes[0], filter.include, {}, function(err, nodes) {
+  self._models[model].model.include(nodes[0], filter.include, options, function(err, nodes) {
```
Comment (jannyHou) [inline lib/connectors/memory.js]: `options || {}` in case the option is `undefined`.

**Comment by raymondfeng** [comment]
@bajtos PTAL

File: lib/utils.js [hunk 2]
```diff
@@ -336,7 +336,7 @@ function sanitizeQuery(query, options) {
   const prohibitedKeys = options.prohibitedKeys;
   const offendingKeys = [];
   const normalizeUndefinedInQuery = options.normalizeUndefinedInQuery;
-  const maxDepth = options.maxDepth || 12;
+  const maxDepth = options.maxDepth || Number.MAX_SAFE_INTEGER;
```
Comment (bajtos) [inline lib/utils.js]: Is it a good idea to allow arbitrary deep queries? Or is this reverting back to the old behavior we used to have in juggler before recent security-related fixes?

**Comment by bajtos** [comment]
> coverage/coveralls — Coverage decreased (-0.01%) to 83.954%

Please check the coverage data, there may be an important use case (test scenario) that you forgot to cover.

File: lib/utils.js (hunk 2 above)
Comment (raymondfeng) [reply lib/utils.js]: This is a default value for the utility method itself. Juggler already sets `32` as the default value.

Comment (bajtos) [reply lib/utils.js]: So can we use `32` as the default value in this place too? I'd rather avoid all possibilities of implicitly infinite depth.

**Comment by raymondfeng** [comment]
> I don't think it's a good idea to move DataAccessObject._normalize into a different file:
The format of Filter object is to specific to DAO only, KeyValue APIs use different filter properties.
It is pretty much impossible to review changes made in that method.
Such huge change will be difficult to back-port to older versions, because there is no automated way how to ensure the backport is not accidentally back-porting changes in _normalize implementation that are not supposed to be back-ported.

The refactoring is to fix https://github.com/strongloop/loopback-datasource-juggler/issues/1663.

**Pull request merged**